basicdataset/src/main/java/ai/djl/basicdataset/nlp/StanfordQuestionAnsweringDataset.java

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.basicdataset.nlp;
+
+import ai.djl.Application.NLP;
+import ai.djl.basicdataset.RawDataset;
+import ai.djl.modality.nlp.embedding.EmbeddingException;
+import ai.djl.ndarray.NDArray;
+import ai.djl.ndarray.NDList;
+import ai.djl.ndarray.NDManager;
+import ai.djl.repository.Artifact;
+import ai.djl.repository.MRL;
+import ai.djl.training.dataset.Record;
+import ai.djl.util.JsonUtils;
+import ai.djl.util.Progress;
+import com.google.gson.reflect.TypeToken;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Stanford Question Answering Dataset (SQuAD) is a reading comprehension dataset, consisting of
+ * questions posed by crowdworkers on a set of Wikipedia articles, where the answer to every
+ * question is a segment of text, or span, from the corresponding reading passage, or the question
+ * might be unanswerable.
+ */
+@SuppressWarnings("unchecked")
+public class StanfordQuestionAnsweringDataset extends TextDataset implements RawDataset<Object> {
```

Comment:
Actually we didn't intend to implement `RawDataset` at first, but later we found SQuAD a quite special dataset because in one record there should be one question and multiple answers (also indices). We encountered a problem while handling labels in `Record` because each embedded answer is a `NDArray`, and we want to have a structure like `NDList(List(answerArray, answerArray, ...), List(indexArray, indexArray))`, but it seems like inside `NDList` there can only be `NDArray`s, so we can't directly implement this structure. We then came up with two resolutions:
1. Stack the `NDArray`s into one `NDArray` so that we can have a big `NDArray` of answers and another for indices. However, this will request the user to break them up into small `NDArray`s if they want to handle the answers separately, and this is very tricky for both the developer and the user.
2. We only return the arrays of the answers and throw the indices away.
After some discussion we chose the second option, which means only the most important data is returned in `get()` method and some other data are lost, which may also be useful, and that's the reason why we decided to implement the `RawDataset` interface to return a JSON object with full data.

Reply:
Actually we didn't intend to implement `RawDataset` at first, but later we found SQuAD a quite special dataset because in one record there should be one question and multiple answers (also indices). We encountered a problem while handling labels in `Record` because each embedded answer is a `NDArray`, and we want to have a structure like `NDList(List(answerArray, answerArray, ...), List(indexArray, indexArray))`, but it seems like inside `NDList` there can only be `NDArray`s, so we can't directly implement this structure. We then came up with two resolutions:

1. Stack the `NDArray`s into one `NDArray` so that we can have a big `NDArray` of answers and another for indices. However, this will request the user to break them up into small `NDArray`s if they want to handle the answers separately, and this is very tricky for both the developer and the user.
2. We only return the arrays of the answers and throw the indices away.

After some discussion we chose the second option, which means only the most important data is returned in `get()` method and some other data are lost, which may also be useful, and that's the reason why we decided to implement the `RawDataset` interface to return a JSON object with full data.